tools/subclass_inferences/materialize_subclass_inferences.py

```diff
@@ -1,70 +1,104 @@
-import os
+"""Generate subClassOf inferences to help OWL-RL reasoners."""
+
+import sys
 import tempfile
-from rdflib import Graph, Namespace, URIRef, Literal
-import rdflib
-from owlready2 import get_ontology, default_world, sync_reasoner
+
+from argparse import ArgumentParser
+
+from rdflib import Graph, Namespace, Literal, BNode
 from rdflib.namespace import RDFS, RDF, SKOS, OWL, XSD
 
-# This script generates gistSubClassAssertions.ttl.
-# From <project-root>/tools directory, run: python subclass_inferences/materialize_subclass_inferences.py
-# The script currently hard-codes Linux path separators, so you need to run from an environment that uses Linux path separators - e.g., Linux commandline, MacOS terminal, or GitBash on Windows.
 
-def run_reasoner(input_ttl, output_ttl):
+try:
+    from owlready2 import get_ontology, default_world, sync_reasoner
+except ImportError:
+    sys.exit('owlready2 reasoner is not in your Python environment, please install')
+
+
+GIST = Namespace("https://w3id.org/semanticarts/ns/ontology/gist/")
+ONTOLOGY = Namespace("https://w3id.org/semanticarts/ontology/")
+
+
+def _add_ontology_declaration(output_graph: Graph, version: str):
+    subclass_ontology = ONTOLOGY.gistSubClassAssertions
+
+    triples = [
+        (RDF.type, OWL.Ontology),
+        (OWL.imports, ONTOLOGY[f"gistCore{version}"]),
+        (OWL.versionIRI, ONTOLOGY[f"gistSubClassAssertions{version}"]),
+        (SKOS.definition, Literal("Supplementary subclass assertions for gistCore.",
+                                  datatype=XSD.string)),
+        (SKOS.prefLabel, Literal("gist Subclass Assertions",
+                                 datatype=XSD.string)),
+        (SKOS.scopeNote, Literal("This ontology contains supplementary subclass "
+                                 "assertions that are logically entailed by gistCore but "
+                                 "are not inferred by some automated reasoners. For example, "
+                                 "an OWL RL reasoner would not infer that gist:Commitment is a "
+                                 "subclass of gist:Intention, although it follows from the "
+                                 "ontology axioms. More precisely, it contains (1) subclass "
+                                 "assertions derived using an OWL DL reasoner and (2) the "
+                                 "subclass assertions that are already explicit in gistCore.",
```

Comment:
```suggestion
        (SKOS.scopeNote, Literal("This ontology contains supplementary subclass "
                                 "assertions to aid some automated reasoners.   For example, "
                                 "while an OWL DL reasoner infers from the class equivalence that "
                                 "gist:Commitment is a subclass of gist:Intention, an RL "
                                 "reasoner does not. More precisely, the ontology contains (1) "
                                 "direct subclass assertions derived using an OWL DL reasoner and "
                                 "(2) the subclass assertions that are already explicit in gistCore.",
```

@sa-bpelakh Can you confirm the following:
* (2) is still true (if not, remove it)
* The indirect subclass triples are filtered out (if not, please remove the word "direct")